requests_lb/requests_lb.py

```diff
@@ -135,34 +136,34 @@ def _retry_request(self, fn, target, **kw):
         Send the given request, as defined by the `fn` argument to the given
         target.
         """
+        _max_request_retries = self._max_request_retries
 
         if target[0] == '/':
             target = target[1:]
 
-        start = self._time()
-
-        while True:
+        while _max_request_retries > 0:
             host_entry = self._srv_next_host()
             (host, port) = host_entry
             url = "{}://{}:{}/{}".format(self._protocol, host, port, target)
 
-            params = dict(kw)
-            params['timeout'] = (self._timeout - (self._time() - start))
-
             try:
                 response = fn(url, **kw)
-            except:
-                log.error("request failed: %s", host_entry,
-                          exc_info=sys.exc_info())
+                response.raise_for_status()
```

Comment:
This should not be necessary, the old code should have allowed `4xx` to pass through to the client, who can then do raise_for_status if they want to. If error codes outside of `503` should be treated as a retry condition they can be done without this (see old code and how `503` was done).